include/rtweekend.hpp

```diff
@@ -7,41 +7,39 @@
 #include <limits>
 #include <memory>
 #include <random>
+#include <type_traits>
 #include <variant>
 #include <vector>
 
 #include <triSYCL/vendor/triSYCL/random/xorshift.hpp>
 
-/// \todo Remove these global objects and move them into the kernel.
-/// It cannot work with SYCL on device otherwise.
-namespace {
-std::uniform_real_distribution<float> distribution(0.0f, 1.0f);
-/// A fast random generator good on accelerator like FPGA
-trisycl::vendor::trisycl::random::xorshift generator;
-} // namespace
+#include <build_parameters.hpp>
 
 // Constants
 
-const float infinity = std::numeric_limits<float>::infinity();
-const float pi = 3.1415926535897932385f;
+constexpr float infinity = std::numeric_limits<float>::infinity();
+constexpr float pi = 3.1415926535897932385f;
```

Comment:
Probably in another PR you can use directly `std::numeric_limits<float_t>::infinity()` where it is used and use PI from C++20 like https://github.com/keryell/muSYCL/blob/master/include/musycl/low_pass_filter.hpp#L46 but adapted for `float_t`